server/app/lib/actions/fusor/activation_key/add_subscriptions.rb

```diff
@@ -36,18 +37,30 @@ def plan(activation_key_id, subscription_descriptions, repositories)
         def run
           ::User.current = ::User.find(input[:user_id])
           key = ::Katello::ActivationKey.find(input[:activation_key_id])
-          associate_subscriptions(key, input[:subscription_descriptions])
+          associate_subscriptions(key, input[:hostgroup])
           enable_repositories(key, input[:repository_cp_labels])
         ensure
           ::User.current = nil
         end
 
         private
 
-        def associate_subscriptions(key, subscription_descriptions)
-          subscription_descriptions.each do |description|
-            subscriptions = key.available_subscriptions.find_all { |sub| sub.description == description }
-            subscriptions.each { |subscription| key.subscribe(subscription.id, 0) } if subscriptions
+        def associate_subscriptions(key, hostgroup)
+          product_name = hostgroup[:activation_key][:content].parameterize.underscore.to_sym
+          products = SETTINGS[:fusor][:content][product_name].map { |p| p[:product_id] }.uniq
+          key.available_subscriptions.find_all.each do |sub|
+            if !sub.products.nil?
+              add = false
+              products.each do |p|
+                if sub.products.find_all { |product| product.cp_id == p }.length > 0
+                  add = true
+                end
+              end
+              if add
+                ::Fusor.log.info "Adding subscription #{sub.id} to activation key #{key.id}"
+                key.subscribe(sub.id, 0)
+              end
+            end
```

Comment:
This logic seems ok, but it seems complicated. So in english, I see the following:

for all of the available subscriptions on the key, see if they have products associated with them.
If they have products, then find the product on the sub that matches the product from fusor.yaml. If you find one,
then we can add this sub.

What I'd really like to see is: find all of the subs whose products match the 'products' from the available_subscriptions.

```
key.available_subscriptions.find_all { |sub| sub.products.cp_id in products}.each do |s|
   key.subscribe(s.id, 0)
end
```
I don't think that's valid syntax at all, but that's what I was thinking. I know I'm over simplifying it. I'll look again to see if I can express this thought in a cleaner way.